### DISCUSSION TITLE
Update `filename` helper such that it doesn't generate filenames that are too long

### DISCUSSION THREAD

**Discussion by mAAdhaTTah** [coddingtonbear/obsidian-web#183] [Ideas]
### Describe the bug

I attempted to clip this URL: https://github.com/OnedocLabs/react-print-pdf using with `{{filename page.title}}.md` as the filename and it fails with EFILETOOLONG.

### To Reproduce

1. Go to https://github.com/OnedocLabs/react-print-pdf
2. Click on the web clipp to attempt a `PUT /vault/{{filename page.title}}.md`
3. See error

### Expected Behavior

My idea would be that the `filename` helper should truncate the filename if it's too long.

### Operating System

OSX

### Browser

Chrome

### Browser Version

123.0.6312.58 (Official Build) (arm64)

**Comment by coddingtonbear** [comment]
This is worth a thought, but I don't know that I'd call it a bug per se.  The role of the `filename` helper is, at least right now, to strip out characters that might be unsafe in a filename -- it's not to guarantee you a _useful_ or _working_ filename, and even if we were to also handle file name length limitations, there are other kinds of limitations that you might still run into (e.g. I believe on Windows the full file path -- which includes the filename -- has to be fewer than 256 chars, and there's no way for us in Obsidian Web to know how many characters are used just to get to where your vault is located).

That all said, I think, that expanding its responsibilities to include truncating at a particular length is alright with me if you were to file a pull request for it.  To do that you'd need to:

- Figure out what the filename length limit must be:
  - It's really likely that Obsidian itself enforces some kind of filename length limitation given that you can sync your vaults across multiple platforms, so if you can figure out what that is and confirm that it's the same on Linux, MacOS, and Windows, that's great, and that's all you need to do at this step.
  - If not, you'll have to find out what the maximum file name length might be across the platforms and filesystems Obsidian vaults might exist on so you can identify what the lowest limit is.  This might include checking at least Windows NTFS, Windows FAT32, MacOs APFS, and Linux EXT4.
- Update https://github.com/coddingtonbear/obsidian-web/blob/master/src/handlebars.ts#L28 to do something reasonable when we calculate a filename that's too long.
- Update the docs here to spell out the new functionality you've written for that helper: https://github.com/coddingtonbear/obsidian-web/wiki/Understanding-Templates#helpers

Cheers & good luck!

**Reply by mAAdhaTTah** [reply]
Ok, cool, I'll do some investigation and see if I can find some time for this. Thanks for the walkthrough!

**Comment by coddingtonbear** [comment]
& as a side note: thanks for accidentally letting me know about https://github.com/OnedocLabs/react-print-pdf -- that looks pretty neat!
